src/main/java/org/generationcp/middleware/dao/GermplasmDAO.java

```diff
@@ -669,7 +669,7 @@ public long countGroupRelatives(final Integer gid) {
 		return toreturn;
 	}
 
-	public List<Germplasm> getChildren(final Integer gid, final char methodType) {
+	public List<Germplasm> getNonGenerativeChildren(final Integer gid, final char methodType) {
```

Comment:
Sorry, let's rename to `getNonGenerativeDescendants`. Also we don't need a `methodType` parameter here. You can remove Germplasm.GET_DERIVATIVE_CHILDREN and Germplasm.GET_MAINTENANCE_CHILDREN and consolidate the query as:
```
"SELECT {g.*}, {n.*} " + "FROM germplsm g LEFT JOIN names n ON g.gid = n.gid AND n.nstat = 1 "
			+ "JOIN methods m ON g.methn = m.mid AND m.mtype IN ('MAN', 'DER') "
			+ "WHERE g.gnpgs = -1 AND g.gpid2 = :gid and  g.deleted = 0  and g.grplce = 0";
```